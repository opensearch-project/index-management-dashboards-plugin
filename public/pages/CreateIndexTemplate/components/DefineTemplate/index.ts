/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import DefineTemplate from "./DefineTemplate";
export { default as OverviewTemplate } from "./OverviewTemplate";

export default DefineTemplate;
