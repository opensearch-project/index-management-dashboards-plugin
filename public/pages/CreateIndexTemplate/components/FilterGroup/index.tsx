import { EuiFilterButton, EuiFilterButtonProps, EuiFilterGroup, EuiFilterSelectItem, EuiLoadingChart, EuiPopover } from "@elastic/eui";
import React, { useState } from "react";
interface IFilterGroupProps {
  options: { label: string }[];
  value?: string[];
  filterButtonProps?: EuiFilterButtonProps;
  onChange?: (val: IFilterGroupProps["value"]) => void;
}

export default function FilterGroup({ options, value, filterButtonProps, onChange }: IFilterGroupProps) {
  const [isPopoverOpen, setIsPopoverOpen] = useState(false);

  const onButtonClick = () => {
    setIsPopoverOpen(!isPopoverOpen);
  };

  const closePopover = () => {
    setIsPopoverOpen(false);
  };

  return (
    <EuiFilterGroup>
      <EuiPopover
        button={
          <EuiFilterButton
            iconType="arrowDown"
            onClick={onButtonClick}
            isSelected={isPopoverOpen}
            numFilters={options?.length}
            hasActiveFilters={!!value?.length}
            numActiveFilters={value?.length}
            {...filterButtonProps}
          />
        }
        isOpen={isPopoverOpen}
        closePopover={closePopover}
        panelPaddingSize="none"
      >
        <div className="ouiFilterSelect__items">
          {options.map((item, index) => (
            <EuiFilterSelectItem
              checked={value?.includes(item.label) ? "on" : "off"}
              key={item.label}
              onClick={() => {
                const findIndex = (value || []).findIndex((valueItem) => valueItem === item.label);
                if (findIndex > -1) {
                  value?.splice(findIndex, 1);
                } else {
                  value?.push(item.label);
                }
                onChange?.([...(value || [])]);
              }}
            >
              {item.label}
            </EuiFilterSelectItem>
          ))}
        </div>
      </EuiPopover>
    </EuiFilterGroup>
  );
}
