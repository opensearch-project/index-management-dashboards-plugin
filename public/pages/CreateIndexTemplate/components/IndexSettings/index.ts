/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import IndexSettings from "./IndexSettings";

export default IndexSettings;
