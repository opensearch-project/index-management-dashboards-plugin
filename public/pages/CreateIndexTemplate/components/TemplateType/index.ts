/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import TemplateType from "./TemplateType";
export { TemplateConvert } from "./TemplateType";

export default TemplateType;
