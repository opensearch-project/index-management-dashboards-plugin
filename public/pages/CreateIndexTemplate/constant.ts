/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export enum TABS_ENUM {
  SUMMARY = "summary",
  CONFIG = "CONFIG",
}
export const tabs = [
  {
    id: TABS_ENUM.SUMMARY,
    name: "Summary",
  },
  {
    id: TABS_ENUM.CONFIG,
    name: "Configuration",
  },
];
