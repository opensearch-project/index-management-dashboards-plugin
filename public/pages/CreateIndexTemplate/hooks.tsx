/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { EuiFormRowProps } from "@elastic/eui";
import { FieldInstance, transformNameToString } from "../../lib/field";

export const getCommonFormRowProps = (name: string | string[], field: FieldInstance): Partial<EuiFormRowProps> => {
  return {
    isInvalid: !!field.getError(name),
    error: field.getError(name),
    "data-test-subj": `form-row-${transformNameToString(name)}`,
  };
};
