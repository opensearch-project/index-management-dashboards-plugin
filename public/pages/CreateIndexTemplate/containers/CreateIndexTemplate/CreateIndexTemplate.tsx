/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { Component, useContext } from "react";
import { RouteComponentProps } from "react-router-dom";
import { isEqual } from "lodash";
import TemplateDetail from "../TemplateDetail";
import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
import { CoreServicesContext } from "../../../../components/core_services";
import { DataSourceMenuContext, DataSourceMenuProperties } from "../../../../services/DataSourceMenuContext";
import { useUpdateUrlWithDataSourceProperties } from "../../../../components/MDSEnabledComponent";
import { getUISettings } from "../../../../services/Services";

interface CreateIndexTemplateProps
  extends RouteComponentProps<{
      template?: string;
      mode?: string;
    }>,
    DataSourceMenuProperties {}

class CreateIndexTemplate extends Component<CreateIndexTemplateProps> {
  static contextType = CoreServicesContext;

  get template() {
    return this.props.match.params.template;
  }

  get readonly() {
    return this.props.match.params.mode === "readonly";
  }

  setBreadCrumb() {
    const readonly = this.readonly;
    let lastBread: typeof BREADCRUMBS.TEMPLATES;
    if (readonly && this.template) {
      lastBread = {
        text: this.template,
        href: `#${this.props.location.pathname}`,
      };
    } else if (this.template) {
      lastBread = {
        text: this.template,
        href: `#${this.props.location.pathname}`,
      };
    } else {
      lastBread = BREADCRUMBS.CREATE_TEMPLATE;
    }
    const uiSettings = getUISettings();
    const useUpdatedUX = uiSettings.get("home:useNewHomePage");
    const breadCrumbs = useUpdatedUX
      ? [BREADCRUMBS.TEMPLATES, lastBread]
      : [BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.TEMPLATES, lastBread];
    this.context.chrome.setBreadcrumbs(breadCrumbs);
  }

  componentDidUpdate(prevProps: Readonly<CreateIndexTemplateProps>): void {
    if (!isEqual(prevProps, this.props)) {
      this.setBreadCrumb();
    }
  }

  componentDidMount = async (): Promise<void> => {
    this.setBreadCrumb();
  };

  onCancel = (): void => {
    this.props.history.push(ROUTES.TEMPLATES);
  };

  render() {
    const uiSettings = getUISettings();
    const useUpdatedUX = uiSettings.get("home:useNewHomePage");

    return !useUpdatedUX ? (
      <div style={{ padding: "0px 50px" }}>
        <TemplateDetail
          history={this.props.history}
          location={this.props.location}
          templateName={this.template}
          onCancel={this.onCancel}
          onSubmitSuccess={() => this.props.history.push(ROUTES.TEMPLATES)}
          dataSourceId={this.props.dataSourceId}
        />
      </div>
    ) : (
      <div style={{ padding: "0px 0px" }}>
        <TemplateDetail
          history={this.props.history}
          location={this.props.location}
          templateName={this.template}
          onCancel={this.onCancel}
          onSubmitSuccess={() => this.props.history.push(ROUTES.TEMPLATES)}
          dataSourceId={this.props.dataSourceId}
        />
      </div>
    );
  }
}

export default function (props: Omit<CreateIndexTemplateProps, keyof DataSourceMenuProperties>) {
  const dataSourceMenuProps = useContext(DataSourceMenuContext);
  useUpdateUrlWithDataSourceProperties();
  return <CreateIndexTemplate {...props} {...dataSourceMenuProps} />;
}
