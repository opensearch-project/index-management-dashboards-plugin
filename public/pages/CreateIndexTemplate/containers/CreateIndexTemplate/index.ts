/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import CreateIndexTemplate from "./CreateIndexTemplate";

export default CreateIndexTemplate;
