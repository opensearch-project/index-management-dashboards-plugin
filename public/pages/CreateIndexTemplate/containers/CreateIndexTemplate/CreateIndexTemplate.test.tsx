/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { MemoryRouter as Router, Route, RouteComponentProps, Switch } from "react-router-dom";
import { render, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CreateIndexTemplate from "./CreateIndexTemplate";
import { ServicesContext } from "../../../../services";
import { browserServicesMock, coreServicesMock, apiCallerMock } from "../../../../../test/mocks";
import { ROUTES } from "../../../../utils/constants";
import { CoreServicesContext } from "../../../../components/core_services";
import { getApplication, getNavigationUI, getUISettings } from "../../../../services/Services";

jest.mock("../../../../services/Services", () => ({
  ...jest.requireActual("../../../../services/Services"),
  getUISettings: jest.fn(),
  getApplication: jest.fn(),
  getNavigationUI: jest.fn(),
}));

beforeEach(() => {
  (getUISettings as jest.Mock).mockReturnValue({
    get: jest.fn().mockReturnValue(false), // or false, depending on your test case
  });
  (getApplication as jest.Mock).mockReturnValue({});

  (getNavigationUI as jest.Mock).mockReturnValue({});
});

function renderCreateIndexTemplateWithRouter(initialEntries = [ROUTES.CREATE_TEMPLATE] as string[]) {
  return {
    ...render(
      <Router initialEntries={initialEntries}>
        <CoreServicesContext.Provider value={coreServicesMock}>
          <ServicesContext.Provider value={browserServicesMock}>
            <Switch>
              <Route
                path={`${ROUTES.CREATE_TEMPLATE}/:template/:mode`}
                render={(props: RouteComponentProps) => <CreateIndexTemplate {...props} />}
              />
              <Route
                path={`${ROUTES.CREATE_TEMPLATE}/:template`}
                render={(props: RouteComponentProps) => <CreateIndexTemplate {...props} />}
              />
              <Route path={ROUTES.CREATE_TEMPLATE} render={(props: RouteComponentProps) => <CreateIndexTemplate {...props} />} />
              <Route path={ROUTES.TEMPLATES} render={(props: RouteComponentProps) => <h1>location is: {ROUTES.TEMPLATES}</h1>} />
            </Switch>
          </ServicesContext.Provider>
        </CoreServicesContext.Provider>
      </Router>
    ),
  };
}

describe("<CreateIndexTemplate /> spec", () => {
  beforeEach(() => {
    apiCallerMock(browserServicesMock);
  });
  it("render template pages", async () => {
    const { findByTitle, container } = renderCreateIndexTemplateWithRouter([`${ROUTES.CREATE_TEMPLATE}/good_template`]);
    await findByTitle("good_template");
    expect(container).toMatchSnapshot();
  });

  it("it goes to indices page when click create successfully in happy path", async () => {
    const { getByText, getByTestId } = renderCreateIndexTemplateWithRouter([`${ROUTES.CREATE_TEMPLATE}`]);

    const templateNameInput = getByTestId("form-row-name").querySelector("input") as HTMLInputElement;
    const submitButton = getByTestId("CreateIndexTemplateCreateButton");
    const shardsInput = getByTestId("form-row-template.settings.index.number_of_shards").querySelector("input") as HTMLInputElement;
    const replicaInput = getByTestId("form-row-template.settings.index.number_of_replicas").querySelector("input") as HTMLInputElement;
    userEvent.type(templateNameInput, `bad_template`);

    userEvent.click(submitButton);
    await waitFor(() => expect(getByText("Index patterns must be defined")).not.toBeNull(), {
      timeout: 3000,
    });

    const patternInput = getByTestId("form-row-index_patterns").querySelector('[data-test-subj="comboBoxSearchInput"]') as HTMLInputElement;
    userEvent.type(patternInput, `test_patterns{enter}`);

    userEvent.click(submitButton);
    await waitFor(() => expect(coreServicesMock.notifications.toasts.addDanger).toBeCalledWith("bad template"));
    userEvent.clear(templateNameInput);
    userEvent.type(templateNameInput, "good_template");

    userEvent.clear(shardsInput);
    userEvent.type(shardsInput, "1.5");
    await waitFor(() => expect(getByText("Number of primary shards must be an integer.")).toBeInTheDocument(), { timeout: 3000 });
    userEvent.clear(shardsInput);
    userEvent.type(shardsInput, "1");

    userEvent.clear(replicaInput);
    userEvent.type(replicaInput, "1.5");
    await waitFor(() => expect(getByText("Number of replicas must be an integer")).toBeInTheDocument(), { timeout: 3000 });
    userEvent.clear(replicaInput);
    userEvent.type(replicaInput, "1");

    userEvent.click(getByTestId("createIndexAddFieldButton"));
    userEvent.click(submitButton);
    await waitFor(() => expect(getByText("Field name is required, please input")).not.toBeNull());
    userEvent.click(getByTestId("mapping-visual-editor-0-delete-field"));

    userEvent.click(submitButton);
    await waitFor(
      () => {
        expect(browserServicesMock.commonService.apiCaller).toBeCalledWith({
          endpoint: "transport.request",
          data: {
            method: "PUT",
            path: "/_index_template/good_template",
            body: {
              _meta: {
                flow: "simple",
              },
              composed_of: [],
              priority: 0,
              template: {
                settings: {
                  "index.number_of_replicas": "1",
                  "index.number_of_shards": "1",
                },
                mappings: {
                  properties: {},
                },
              },
              index_patterns: ["test_patterns"],
            },
          },
        });
        expect(getByText(`location is: ${ROUTES.TEMPLATES}`)).toBeInTheDocument();
      },
      {
        timeout: 3000,
      }
    );
  }, 20000);
});
