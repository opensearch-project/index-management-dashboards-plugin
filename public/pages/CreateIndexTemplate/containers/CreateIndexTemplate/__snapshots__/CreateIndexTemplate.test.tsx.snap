// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateIndexTemplate /> spec render template pages 1`] = `
<div>
  <div
    style="padding: 0px 50px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <h1
          class="euiTitle euiTitle--large"
          title="good_template"
        >
          good_template
        </h1>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
        style="flex-direction: row;"
      >
        <button
          class="euiButton euiButton--primary euiButton--small"
          style="margin-right: 20px;"
          type="button"
        >
          <span
            class="euiButtonContent euiButton__content"
          >
            <span
              class="euiButton__text"
            >
              View JSON
            </span>
          </span>
        </button>
        <button
          class="euiButton euiButton--danger euiButton--small"
          type="button"
        >
          <span
            class="euiButtonContent euiButton__content"
          >
            <span
              class="euiButton__text"
            >
              Delete
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    >
      <div
        class="euiText euiText--small"
      >
        <h2>
          Overview
        </h2>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
      />
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div
        class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--fourths euiFlexGrid--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row euiDescriptionList--compressed"
          >
            <dt
              class="euiDescriptionList__title"
            >
              <div
                class="euiText euiText--small"
              >
                <h4>
                  Template type
                </h4>
              </div>
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              Indexes
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row euiDescriptionList--compressed"
          >
            <dt
              class="euiDescriptionList__title"
            >
              <div
                class="euiText euiText--small"
              >
                <h4>
                  Index patterns
                </h4>
              </div>
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              -
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row euiDescriptionList--compressed"
          >
            <dt
              class="euiDescriptionList__title"
            >
              <div
                class="euiText euiText--small"
              >
                <h4>
                  Priority
                </h4>
              </div>
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              undefined
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row euiDescriptionList--compressed"
          >
            <dt
              class="euiDescriptionList__title"
            >
              <div
                class="euiText euiText--small"
              >
                <h4>
                  Associated component templates
                </h4>
              </div>
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              -
            </dd>
          </dl>
        </div>
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      class="euiTabs euiTabs--small"
      role="tablist"
    >
      <button
        aria-selected="true"
        class="euiTab euiTab-isSelected"
        data-test-subj="TemplateDetailTab-summary"
        role="tab"
        type="button"
      >
        <span
          class="euiTab__content"
        >
          Summary
        </span>
      </button>
      <button
        aria-selected="false"
        class="euiTab"
        data-test-subj="TemplateDetailTab-CONFIG"
        role="tab"
        type="button"
      >
        <span
          class="euiTab__content"
        >
          Configuration
        </span>
      </button>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsFlexStart euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        style="margin-top: 0px; margin-bottom: 0px; padding: 0px 10px;"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiText euiText--small"
          >
            <h2>
              Preview template
            </h2>
          </div>
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full"
      />
      <div
        style="padding: 0px 10px;"
      >
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
          id="some_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="some_html_id"
            >
              <div
                class="euiText euiText--small"
              >
                <h3>
                  Index alias
                </h3>
              </div>
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormHelpText euiFormRow__text"
              style="padding-top: 0px; padding-bottom: 4px;"
            >
              Allow the new indexes to be referenced by existing aliases or specify a new alias.
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--fourths euiFlexGrid--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row euiDescriptionList--compressed"
            >
              <dt
                class="euiDescriptionList__title"
              >
                <div
                  class="euiText euiText--small"
                >
                  <h4>
                    Alias names
                  </h4>
                </div>
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--l"
        />
        <div
          class="euiText euiText--small"
        >
          <h3>
            Index settings
          </h3>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--fourths euiFlexGrid--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row euiDescriptionList--compressed"
            >
              <dt
                class="euiDescriptionList__title"
              >
                <div
                  class="euiText euiText--small"
                >
                  <h4>
                    Number of primary shards
                  </h4>
                </div>
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                1
              </dd>
            </dl>
          </div>
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row euiDescriptionList--compressed"
            >
              <dt
                class="euiDescriptionList__title"
              >
                <div
                  class="euiText euiText--small"
                >
                  <h4>
                    Number of replicas
                  </h4>
                </div>
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                10
              </dd>
            </dl>
          </div>
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row euiDescriptionList--compressed"
            >
              <dt
                class="euiDescriptionList__title"
              >
                <div
                  class="euiText euiText--small"
                >
                  <h4>
                    Refresh interval
                  </h4>
                </div>
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--l"
        />
        <div
          class="euiAccordion accordion-in-advanced-settings"
        >
          <div
            class="euiAccordion__triggerWrapper"
          >
            <button
              aria-controls="accordionForCreateIndexTemplateSettings"
              aria-expanded="false"
              class="euiAccordion__button"
              id="some_html_id"
              type="button"
            >
              <span
                class="euiAccordion__iconWrapper"
              >
                EuiIconMock
              </span>
              <span
                class="euiIEFlexWrapFix"
              >
                <h4>
                  Advanced settings
                </h4>
              </span>
            </button>
          </div>
          <div
            aria-labelledby="some_html_id"
            class="euiAccordion__childWrapper"
            id="accordionForCreateIndexTemplateSettings"
            role="region"
            style="height: 0px"
            tabindex="-1"
          >
            <div>
              <div
                class=""
              >
                <div
                  class="euiSpacer euiSpacer--s"
                />
                <div
                  class="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
                  id="some_html_id-row"
                >
                  <div
                    class="euiFormRow__labelWrapper"
                  >
                    <label
                      class="euiFormLabel euiFormRow__label"
                      for="some_html_id"
                    >
                      Specify advanced index settings
                    </label>
                  </div>
                  <div
                    class="euiFormRow__fieldWrapper"
                  >
                    <div
                      class="euiFormHelpText euiFormRow__text"
                      style="padding-top: 0px; padding-bottom: 4px;"
                    >
                      <p>
                        Specify a comma-delimited list of settings.
                         
                        <a
                          class="euiLink euiLink--primary"
                          href="https://opensearch.org/docs/latest/api-reference/index-apis/create-index#index-settings"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          View index settings
                          EuiIconMock
                          <span
                            class="euiScreenReaderOnly"
                          >
                            (opens in a new tab or window)
                          </span>
                        </a>
                      </p>
                      <p>
                        All the settings will be handled in flat structure.
                         
                        <a
                          class="euiLink euiLink--primary"
                          href="https://opensearch.org/docs/latest/api-reference/index-apis/get-index/#url-parameters"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          Learn more
                          EuiIconMock
                          <span
                            class="euiScreenReaderOnly"
                          >
                            (opens in a new tab or window)
                          </span>
                        </a>
                      </p>
                    </div>
                    <textarea
                      data-test-subj="jsonEditor-valueDisplay"
                      readonly=""
                      style="display: none;"
                    >
                      {
  "index.number_of_replicas": "10",
  "index.number_of_shards": "1"
}
                    </textarea>
                    <div
                      class="euiCodeEditorWrapper"
                      data-test-subj="codeEditorContainer"
                      style="width: 100%;"
                    >
                      <button
                        class="euiCodeEditorKeyboardHint"
                        data-test-subj="codeEditorHint"
                        id="some_html_id"
                      >
                        <p
                          class="euiText"
                        >
                          Press Enter to start editing.
                        </p>
                        <p
                          class="euiText"
                        >
                          When you're done, press Escape to stop editing.
                        </p>
                      </button>
                      <div
                        class=" ace_editor ace-tm"
                        id="some_html_id"
                        style="width: 100%; height: 500px;"
                      >
                        <textarea
                          autocapitalize="off"
                          autocorrect="off"
                          class="ace_text-input"
                          spellcheck="false"
                          style="opacity: 0;"
                          tabindex="-1"
                          wrap="off"
                        />
                        <div
                          aria-hidden="true"
                          class="ace_gutter"
                        >
                          <div
                            class="ace_layer ace_gutter-layer ace_folding-enabled"
                          />
                          <div
                            class="ace_gutter-active-line"
                          />
                        </div>
                        <div
                          class="ace_scroller"
                        >
                          <div
                            class="ace_content"
                          >
                            <div
                              class="ace_layer ace_print-margin-layer"
                            >
                              <div
                                class="ace_print-margin"
                                style="left: 4px; visibility: visible;"
                              />
                            </div>
                            <div
                              class="ace_layer ace_marker-layer"
                            />
                            <div
                              class="ace_layer ace_text-layer"
                              style="padding: 0px 4px;"
                            />
                            <div
                              class="ace_layer ace_marker-layer"
                            />
                            <div
                              class="ace_layer ace_cursor-layer ace_hidden-cursors"
                            >
                              <div
                                class="ace_cursor"
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="ace_scrollbar ace_scrollbar-v"
                          style="display: none; width: 20px;"
                        >
                          <div
                            class="ace_scrollbar-inner"
                            style="width: 20px;"
                          />
                        </div>
                        <div
                          class="ace_scrollbar ace_scrollbar-h"
                          style="display: none; height: 20px;"
                        >
                          <div
                            class="ace_scrollbar-inner"
                            style="height: 20px;"
                          />
                        </div>
                        <div
                          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
                        >
                          <div
                            style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
                          />
                          <div
                            style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
                          >
                            XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--l"
        />
        <div
          class="euiText euiText--small"
        >
          <h3>
            Index mapping
          </h3>
        </div>
        <div
          class="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
          id="some_html_id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormHelpText euiFormRow__text"
              id="some_html_id-help-0"
            >
              <div>
                Define how documents and their fields are stored and indexed.
                 
                <a
                  class="euiLink euiLink--primary"
                  href="https://opensearch.org/docs/undefined/opensearch/mappings/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Learn more
                  EuiIconMock
                  <span
                    class="euiScreenReaderOnly"
                  >
                    (opens in a new tab or window)
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
          id="some_html_id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <fieldset
              class="euiButtonGroup euiButtonGroup--small euiButtonGroup--text"
            >
              <legend
                class="euiScreenReaderOnly"
              >
                Editor Type
              </legend>
              <div
                class="euiButtonGroup__buttons"
              >
                <label
                  class="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small euiButtonGroupButton--fill euiButtonGroupButton-isSelected"
                  data-test-subj="editorTypeVisualEditor"
                  for="some_html_id"
                >
                  <span
                    class="euiButtonContent euiButton__content"
                  >
                    <span
                      class="euiButton__text euiButtonGroupButton__textShift"
                      data-text="Tree view"
                      title="Tree view"
                    >
                      <input
                        checked=""
                        class="euiScreenReaderOnly"
                        data-test-subj="VISUAL"
                        id="some_html_id"
                        name="some_html_id"
                        type="radio"
                        value=""
                      />
                      Tree view
                    </span>
                  </span>
                </label>
                <label
                  class="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small"
                  data-test-subj="editorTypeJsonEditor"
                  for="some_html_id"
                >
                  <span
                    class="euiButtonContent euiButton__content"
                  >
                    <span
                      class="euiButton__text euiButtonGroupButton__textShift"
                      data-text="JSON"
                      title="JSON"
                    >
                      <input
                        class="euiScreenReaderOnly"
                        data-test-subj="JSON"
                        id="some_html_id"
                        name="some_html_id"
                        type="radio"
                        value=""
                      />
                      JSON
                    </span>
                  </span>
                </label>
              </div>
            </fieldset>
            <div
              class="euiSpacer euiSpacer--l"
            />
            <div
              class="euiText euiText--small"
            >
              <p>
                You have no field mappings.
              </p>
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--l"
        />
      </div>
    </div>
  </div>
</div>
`;
